app/mongo_db_connection.py

```diff
@@ -1170,3 +1170,43 @@ def delete_items_in_folder(item_id, folder_id, item_type):
         }
     )
     return post_to_data_service(payload)
+
+def bulk_query_template_collection_modified(options, status=None):
+    """
+    Fetch templates based on specified options and status.
+
+    Parameters:
+        options (dict): Additional query parameters to filter templates.
+        status (str): The status of templates to filter ("save templates" or "draft").
+
+    Returns:
+        list: A list of templates that match the specified options and status.
+    """
+    templates = get_data_from_data_service(*TEMPLATE_CONNECTION_LIST, "fetch", field=options)
+
+    if status is not None:
+        filtered_templates = [template for template in templates if template.get("template_state") == status]
+        return filtered_templates
+
+    return templates
+
+
+
+def bulk_query_document_collection_modified(options, status=None):
```

Comment:
This is what we have been avoiding, Stick to using the bulk query template collection. pass status in the options object.